daemons/gptp/common/ptp_message.cpp

```diff
@@ -959,9 +962,8 @@ void PTPMessageFollowUp::processMessage(IEEE1588Port * port)
 	master_local_freq_offset += 1.0;
```

Comment:
I have worked through the various ratios, and I believe this change is correct.

However, I think I found another existing bug in line 961; the equation from the specification is `cumulativeScaledRateOffset * 2^(-41)`, which means we should be dividing by `1ULL << 41` rather than `2ULL << 41`.

Since `FrequencyRatio` is `long double`, I guess we'll have enough precision leftover to deal with adding 1.0 (this effectively shifts 32 bits worth of mantissa precision to the right by 41 bits as we zero the exponent) but we'll still lose ~9 bits if 80-bit x86 extended precision floats are being used.  We could run into trouble in the future with these ratio calculations on systems where `long double` is the same thing as `double`, though; there are only effectively 53 bits of precision available there, so we'll lose 20 out of our original 32 bits.